src/list-sorting/index.js

```diff
@@ -1,3 +1,27 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack)
+{
+  let lastIndex = -1;
+  /* we traverse backwards in order to spot the last appearance first,
+  hence the use of - 1 to account for zero-indexing */
+  for (let i = haystack.length; i > 0; i--)
+  {
+    //if the array item is iterable it is a 2D array
+    if (typeof haystack[i - 1][Symbol.iterator] === 'function')
```

Comment:
This is overkill for `Array.isArray`. Your `completeness` would take a hit if you do this, see your rubric.